.npmrc

```diff
@@ -0,0 +1,14 @@
+hoist-pattern[]=!**/typechain/*
+hoist-pattern[]=!**/@typechain/*
+hoist-pattern[]=!@eth-optimism/contracts-bedrock/ds-test
+hoist-pattern[]=!@eth-optimism/contracts-bedrock/forge-std
+hoist-pattern[]=!@eth-optimism/contracts-bedrock/@rari-capital/solmate
+hoist-pattern[]=!@openzeppelin
+hoist-pattern[]=!@eth-optimism/contracts-periphery/ds-test
+hoist-pattern[]=!@eth-optimism/contracts-periphery/forge-std
+hoist-pattern[]=!@eth-optimism/contracts-periphery/@rari-capital/solmate
+hoist-pattern[]=!forta-agent
+
+# run prefoo scripts
+# npm and yarn support this as a default pnpm defaults to false
+enable-pre-post-scripts=true
```

Comment:
We should remove this behavior [in favor of nx](https://github.com/ethereum-optimism/optimism/pull/4778) controlling the order that things need to be built

Reply:
My nx pr removes all the prebuild pretest prelint ect. in favor of nx

Reply:
Not shamefully hoisting here means we need to test anything that doesn't have test coverage